pkg/apis/openebs.io/upgrade/v1alpha1/upgraderesult.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	// Register adds UpgradeResult and UpgradeResultList objects to
+	// SchemeBuilder so they can be added to a Scheme
+	SchemeBuilder.Register(&UpgradeResult{}, &UpgradeResultList{})
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=upgrade
+
+// UpgradeResult contains the desired specifications of an
+// upgrade result
+type UpgradeResult struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Config UpgradeResultConfig `json:"config"`
+	Status UpgradeResultStatus `json:"status"`
```

Comment:
Will it be good to have Tasks here?
```go
type UpgradeResult struct {
  ...
  ...
  Config  UpgradeResultConfig  `json:"config"`
  Tasks   UpgradeResultTask    `json:"tasks"`
  Status  UpgradeResultStatus  `json:"status"`
}

Reply:
I believe tasks should be part of the resources itself since it would be helpful to refer to the runtasks of a particular resource say runtasks of pool A or B.

Reply:
@sagarkrsd it might be good. it seems to be good as we can cross refer easily.
However, i wanted to delink the logic part from the status part.
e.g. a task can have some retries, errorCount, successCount, time taken & so on. However this are logic specific.
The resources & subresources should be restricted to their schema i.e. display preState, postState.
The resources & sub-resources should show their CR state only. we should not set the status of a resource with task executor message.

Reply:
Thanks @AmitKumarDas , will do that.